recipes/m4/all/test_package/conanfile.py

```diff
@@ -0,0 +1,20 @@
+from conans import ConanFile
+import os
+
+
+class TestPackageConan(ConanFile):
+
+    def build(self):
+        test_file = os.path.join(self.source_folder, "test_package.m4")
+        in_file = os.path.join(self.build_folder, "input.m4")
+        with open(in_file, "w") as f_in_file, open(test_file, "r") as f_test_file:
+            f_in_file.write(f_test_file.read().replace("\r\n", "\r"))
+        assert "\r\n" not in open(in_file, "r")
+
+        self.run("{} --version".format(os.environ["M4"]))
+        self.run("{} -P {}".format(os.environ["M4"], in_file))
+        self.run("{} -P {} > out".format(os.environ["M4"], in_file))
+
+    def test(self):
+        if "\r\n" in open("out", "rb"):
```

Comment:
@SSE4 My fix for the test_package was wrong because I am mixing text mode and binary mode.

Reply:
I'll open a pr

Reply:
bincrafters/conan-center-index#5

No fix for bison yet.
The unit test of m4 fail when using Visual Studio.

Reply:
The unit tests on Windows are mostly failing due to mismatching paths and not being ably to run some linux tools.
- Expected paths do not match actual paths. For example `m4.exe:stdin:1: input error` does not match `C:\users\maarten\.conan\data\m4\1.4.18\_\_\build\456f15897172eef340fcbac8a70811f2beb26a93\src\m4.exe:stdin:1: input error`
- For some reason the tests are unable to run `echo` using `esyscmd`.

These errors also fail when using mingw. And since m4 with bison is working, these failing tests are a false positive.
Back to square one.